File: Program's_Contributed_By_Contributors/Golang Programs/linear_search.go
package main
import "fmt"

func linearSearch(A []int, n int, x int) {
 for i := 0; i < n; i++ {
  if A[i] == x {
   fmt.Println("Number found at position : ", i+1)
   return
  }
 }

 fmt.Println("Number not found")
 return

}

func main() {
 fmt.Println("Enter the size of the array")
 var n int
 fmt.Scan(&n)

 var x int
 A := make([]int, n, 100)
fmt.Println("Enter elements of the array : ")

 for i := 0; i < n; i++ {
  fmt.Scan(&A[i])
 }

 fmt.Println("Enter the number to be searched")
 fmt.Scan(&x)

 linearSearch(A, n, x)

}

File: Program's_Contributed_By_Contributors/c language/linear_search.c
#include<stdio.h>
int main()
{
	int arr[20],n,i,key,result;
	printf("enter the number of elements in an array\n");
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		scanf("%d",&arr[i]);
	}
	printf("enter the element to be searched");
	scanf("%d",&key);
	result=linear_search(arr,n,key);
	if(result!=-1)
	printf("found at position %d",result);
	else
	printf("Unsuccesful Search");
	return 0;
}
int linear_search(int a[],int n,int k)
{
	int i,pos;
	for(i=0;i<n;i++)
	{
		if(a[i]==k)
		{
			pos=i;
			return pos+1;
		}
	}
		return -1;
}
